File: cpp/class_inheritance.cpp
#include<iostream>
using namespace std;
class Vehicle
{
	private :
	char color[15];
	

public:
	int wheels;
	Vehicle(){

	}
	void start(){
		cout << "Vehicle started\n";
	}
	void stop()
	{
		cout << "Vehicle stop\n";
	}
	void info(){
		cout << "vehicle info\n";
	}
};



class Bike : public Vehicle // inheritance
{
	
public:
	Bike(){
		wheels=2;
		cout << "inside bike constructor\n";

	}

	void start(){
		cout << "Bike started\n";
	}
	void stop()
	{
		cout << "Bike stop\n";
	}

	void info(){
		cout<<"Bike info wheels "<<wheels << "\n";
	}
};
class Car : public Vehicle  // inheritance
{
public:
	Car(){
		int	wheels=4;
	}
	void start(){
		cout << "Car started\n";
	}
	void stop()
	{
		cout << "Car stop\n";
	}
 
};
int main()
{

	Vehicle v = Vehicle(); // creating object of a class
    v.start();
	v.info();
	v.stop();


	Bike b = Bike();
	b.start();
	b.info();
	b.stop();


	Car c = Car();
	c.start();
	c.info();
	c.stop();

   
    Vehicle v1 = Bike();
    // Bike b1 = Vehicle(); not allowed creating a child object from parent



	return 0;
}

File: python/class_inheritance.py
class Vehicle:

	def __init__(self):
		pass

	def start(self):
		print("Vehicle started\n")
	
	def stop(self):
	
		print("Vehicle stop\n")
	
	def info(self):
		print("vehicle info\n")
	


class Bike(Vehicle): # inheritance
	def __init__(self):
		self.wheels=2
		print("inside bike constructor\n")

	def start(self):
		print("Bike started\n")
	
	def stop(self):
		print("Bike stop\n")
	

	def info(self):
		print(f"Bike info wheels {self.wheels}\n")
	
class Car(Vehicle): # inheritance
	def __init__(self):
		self.wheels=4
		print("inside Car constructor\n")

	def start(self):
		print("Car started\n")
	
	def stop(self):
		print("Car stop\n")
	
#main
v = Vehicle() #creating object of a class
v.start()
v.info()
v.stop()


b = Bike()
b.start()
b.info()
b.stop()


c = Car()
c.start()
c.info()
c.stop()


v1 = Bike()
#b1 = Vehicle(); not allowed creating a child object from parent

